### DISCUSSION TITLE
Warnings/errors on duplicate lifecycle phases

### DISCUSSION THREAD

**Discussion by marcphilipp** [support-and-care/maven-support-and-care#14] [Ideas for Funding]
When invoking Maven with redundant lifecycle phases, e.g. `mvn clean package verify install`, goals are executed multiple times even though that's rarely what was intended. In the example, it was probably `mvn clean install` so one option would be to issue a warning and execute that instead. Another option would be to fail such invocations to educate developers that they're wasting time and resources and tell them what to do instead.

**Comment by slawekjaranowski** [comment]
https://issues.apache.org/jira/browse/MNG-5885

**Comment by khmarbaise** [comment]
We should fail that build and give hints what should be done in a better way.

**Comment by ascheman** [comment]
This discussion has been converted to Epic #50 for tracking.

Closing this discussion.
